samples/README.md

```diff
@@ -0,0 +1,27 @@
+# Samples, Snippets, and How-To Guides
+
+Developers like to learn by looking at code, and so the Azure SDK comes with many samples in the form of short code snippets, sample applications, and how-to guides. This document describes where to find all these resources.
+
+## Structure of the Repository
+The Azure SDK repository is organized in the following folder structure, with the main sample locations highlighted using **bold** font.
+
+- `/samples` (this folder)
+  - `README.md` (this file)
+- `/sdk` (folder containing sources for all SDK packages)
+  - `/<service>` (e.g. storage)
+    - `/<package>` (e.g. `@azure/storage-blob`)
+      - **`README.md`** (package READMEs contain "hello world" code snippets)
+      - **`/samples`**
+        - **`/typescript`** (package-specific samples in TypeScript)
+        - **`/javascript`** (package-specific samples in JavaScript using CommonJS)
+      - `/src`
+      - `/tests`
+
+##  Getting Started (a.k.a. `Hello World`) Samples
+Each package folder contains a package-specific `README.md` file. Most of these `README` files contain `Hello World` code samples illustrating basic usage of the the APIs contained in the package. For example, you can find `Hello World` samples for the `@azure/storage-blob` package [here](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/storage/storage-blob#examples).
+
+## Package Samples and How-To Guides
+Each package folder contains a subfolder called `samples` with additional code samples in the `typescript` and `javascript` subfolder. Corresponding files (i.e. `samples/typescript/src/helloWorld.ts` and `samples/javascript/helloWorld.js`) in the TypeScript and JavaScript directories are functionally equivalent. These samples may be viewed in GitHub or browsed on [the Sample Browser at docs.microsoft.com/samples](https://docs.microsoft.com/samples/browse/?products=azure&languages=typescript%2Cjavascript) by selecting the product and language combination that you wish to view samples for.
+
+## Sample Applications
```

Comment:
do we want to call out the bundling samples? 

https://github.com/Azure/azure-sdk-for-js/tree/master/samples/Bundling

It's the only non-package specific example we have right now, but I can imagine adding a couple more in the future to do generic things (e.g. how to test with our libs)

Reply:
yeah good point I'll make a quick change to this.

Reply:
can you also add a link to the docs for this? https://github.com/Azure/azure-sdk-for-js/blob/master/documentation/Bundling.md